### PR TITLE
CATALOG-9219: [update] Catalog, update all Product Custom Fields endpoints

### PR DESCRIPTION
**Pull request by olysenko**
# CATALOG-9219 Review and update public API documentation for the POST|GET|PUT|DELETE v3/catalog/products/:productId/custom-fields

## What changed?
* Updated v3 custom-fields swagger scheme

## Release notes draft
* updated error messaging for custom fields;
* clarified info about limitation of 250 max allowed characters for the name and value in custom fields.

ping @bigcommerce/team-catalog

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -6,3 +6,10 @@
 # be identified in the format @org/team-name. Teams must have
 # explicit write access to the repository. In this example,
 # the octocats team in the octo-org organization owns all .md files.
+
+###
+# Team Specific
+###
+
+# @bigcommerce/team-catalog
+reference/catalog @bigcommerce/team-catalog
```
Comment (slsriehl) [inline .github/CODEOWNERS:15]: @olysenko we're not to the point of assigning domain-specific codeowners. please remove.

File: reference/catalog/custom-fields.v2.yml [hunk 2]
```diff
@@ -0,0 +1,390 @@
+openapi: '3.0.3'
```
Comment (slsriehl) [inline reference/catalog/custom-fields.v2.yml:1]: @olysenko we do not currently provide any swagger for the Catalog V2 API. let's discuss with y'all, @bc-tgomez, and myself.

File: reference/catalog/custom-fields.v3.yml [hunk 3]
```diff
@@ -0,0 +1,479 @@
+openapi: '3.0.3'
```
Comment (slsriehl) [inline reference/catalog/custom-fields.v3.yml:1]: @olysenko what's the rationale for removing Catalog Product Custom Fields from the Catalog Products file? Let's discuss.

File: reference/catalog/custom-fields.v3.yml [hunk 4]
```diff
@@ -0,0 +1,479 @@
...
+              example: 36
+            count:
+              type: integer
+              description: |
+                Total number of items in the collection response.
+              example: 36
+            per_page:
+              type: integer
+              description: |
+                The amount of items returned in the collection per page, controlled by the limit parameter.
```
Comment (bc-traciporter) [inline reference/catalog/custom-fields.v3.yml:289]:
```suggestion
                The number of items returned in the collection per page. The limit parameter controls this field.
```

File: reference/catalog/custom-fields.v3.yml [hunk 5]
```diff
@@ -0,0 +1,479 @@
...
+        - errors
+      title: Error Response
+      type: object
+      properties:
+        status:
+          type: integer
+          description: The HTTP status code.
+        title:
+          type: string
+          description: The error title describing the particular error.
```
Comment (bc-traciporter) [inline reference/catalog/custom-fields.v3.yml:343]:
```suggestion
          description: The error title describes the particular error.
```

File: reference/catalog/custom-fields.v3.yml [hunk 6]
```diff
@@ -0,0 +1,479 @@
...
+        - type
+      title: Error Response
+      type: object
+      properties:
+        status:
+          type: integer
+          description: The HTTP status code.
+        title:
+          type: string
+          description: The error title describing the particular error.
```
Comment (bc-traciporter) [inline reference/catalog/custom-fields.v3.yml:364]:
```suggestion
          description: The error title describes the particular error.
```

File: reference/catalog/custom-fields.v3.yml [hunk 7]
```diff
@@ -0,0 +1,479 @@
...
+      title: Error Response
+      type: object
+      properties:
+        status:
+          type: integer
+          description: The HTTP status code.
+          example: 405
+        title:
+          type: string
+          description: The error title describing the particular error.
```
Comment (bc-traciporter) [inline reference/catalog/custom-fields.v3.yml:385]:
```suggestion
          description: The error title describes the particular error.
```

File: reference/catalog/custom-fields.v3.yml [hunk 8]
```diff
@@ -0,0 +1,479 @@
...
+        type:
+          type: string
+        detail:
+          title: Detailed Errors
+          type: string
+          description: The details title describing the particular error.
```
Comment (bc-traciporter) [inline reference/catalog/custom-fields.v3.yml:391]:
```suggestion
          description: The detailed title describes the particular error.
```

File: reference/catalog/custom-fields.v3.yml [hunk 9]
```diff
@@ -0,0 +1,479 @@
...
+          schema:
+           properties:
+              data:
+               type: array
+               items:
+                $ref: '#/components/schemas/customFieldData'
+              meta:
+                $ref: '#/components/schemas/metaCollectionFull'
+    CustomFieldResponse:
+      description: Gets Custom field
```
Comment (bc-traciporter) [inline reference/catalog/custom-fields.v3.yml:436]:
```suggestion
      description: Gets Custom field.
```

File: reference/catalog/custom-fields.v3.yml [hunk 10]
```diff
@@ -0,0 +1,479 @@
...
+          schema:
+            $ref: '#/components/schemas/MethodNotAllowedError'
+    UnsupportedMediaTypeError:
+      description: 415 Unsupported Media Type
+      content:
+        text/html:
+          schema:
+            type: string
+    CustomFieldsResponse:
+      description: Gets array of Custom fields
```
Comment (bc-traciporter) [inline reference/catalog/custom-fields.v3.yml:424]:
```suggestion
      description: Gets array of Custom fields.
```

**Comment by slsriehl** [comment]
[DEVDOCS-5771] @bc-tgomez

**Review by bc-traciporter — COMMENTED** [review]
I left editorial suggestions. I mainly tried to remove passive voice.  Feel free to not accept my suggestion if the change does not convey the correct meaning.

File: .github/CODEOWNERS (hunk 1 above)
Comment (olysenko) [reply .github/CODEOWNERS:15]: OK. I will remove this change.

File: reference/catalog/products_catalog.v3.yml [hunk 11]
```diff
@@ -4157,412 +4106,62 @@ paths:
         - name
         - value
 
+        **Name-Value Pair Uniqueness**
+        - every name-value pair must be unique inside one product
```
Comment (bc-traciporter) [inline reference/catalog/products_catalog.v3.yml:4110]:
```suggestion
        - Every name-value pair must be unique inside a product.
```

File: reference/catalog/products_catalog.v3.yml [hunk 12]
```diff
@@ -9226,6 +8759,248 @@ components:
...
+        - name
+        - value
+      type: object
+      properties:
+        name:
+          maxLength: 250
+          minLength: 1
+          type: string
+          description: |
+            The name of the field, shown on the storefront, orders, etc. Required for /POST
```
Comment (bc-traciporter) [inline reference/catalog/products_catalog.v3.yml:8799]:
```suggestion
            The name of the field, shown on the storefront, orders, etc. This field is a requirement for /POST requests.
```

File: reference/catalog/products_catalog.v3.yml [hunk 13]
```diff
@@ -9226,6 +8759,248 @@ components:
...
+          example: 'ISBN'
+        value:
+          maxLength: 250
+          minLength: 1
+          type: string
+          description: |
+            The value of the field, shown on the storefront, orders, etc. Required for /POST
```
Comment (bc-traciporter) [inline reference/catalog/products_catalog.v3.yml:8806]:
```suggestion
            The value of the field, shown on the storefront, orders, etc. This field is a requirement for /POST requests.
```

File: reference/catalog/products_catalog.v3.yml [hunk 14]
```diff
@@ -9226,6 +8759,248 @@ components:
...
+    customFieldPut:
+      title: Custom Field Put
+      type: object
+      properties:
+        name:
+          maxLength: 250
+          minLength: 1
+          type: string
+          description: |
+            The name of the field, shown on the storefront, orders, etc. Required for /POST
```
Comment (bc-traciporter) [inline reference/catalog/products_catalog.v3.yml:8818]:
```suggestion
            The name of the field, shown on the storefront, orders, etc. This field is a requirement for /POST requests.
```

File: reference/catalog/products_catalog.v3.yml [hunk 15]
```diff
@@ -9226,6 +8759,248 @@ components:
...
+          example: 'ISBN'
+        value:
+          maxLength: 250
+          minLength: 1
+          type: string
+          description: |
+            The value of the field, shown on the storefront, orders, etc. Required for /POST
```
Comment (bc-traciporter) [inline reference/catalog/products_catalog.v3.yml:8825]:
```suggestion
            The value of the field, shown on the storefront, orders, etc. This field is a requirement for /POST requests.

**Review by bc-traciporter — COMMENTED** [review]
Left a few suggestions

**Review by bc-traciporter — DISMISSED** [review]
LGTM!

**Review by bc-traciporter — APPROVED** [review]
LGTM

**Pull request merged**